jobs/mongodb_backup_job.go

```diff
@@ -147,19 +186,92 @@ func createDBURL(dbConfig DBConnConfig) *url.URL {
 	}
 }
 
-func (j *MongoDBBackupJob) startBackup(ctx context.Context) error {
+func (j *MongoDBBackupJob) startBackup(ctx context.Context) (pbmBackup, error) {
 	j.l.Info("Starting backup.")
 
 	nCtx, cancel := context.WithTimeout(ctx, cmdTimeout)
 	defer cancel()
 
-	output, err := exec.CommandContext(nCtx, pbmBin, "backup", "--mongodb-uri="+j.dbURL.String()).CombinedOutput() // #nosec G204
+	var stderr bytes.Buffer
+	var stdout bytes.Buffer
+	cmd := exec.CommandContext(nCtx, pbmBin, "backup", "--out=json", "--mongodb-uri="+j.dbURL.String()) // #nosec G204
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return pbmBackup{}, errors.Wrapf(err, "pbm backup error: %s", stderr.String())
+	}
 
-	if err != nil {
-		return errors.Wrapf(err, "pbm backup error: %s", string(output))
+	var result pbmBackup
+	if err := json.NewDecoder(&stdout).Decode(&result); err != nil {
+		return pbmBackup{}, err
 	}
 
-	return nil
+	return result, nil
+}
+
+func (j *MongoDBBackupJob) streamLogs(ctx context.Context, send Send, name string) error {
+	ticker := time.NewTicker(logsCheckInterval)
+	defer ticker.Stop()
+	var buffer bytes.Buffer
+	skip := 0
+	for {
+		select {
+		case <-ticker.C:
+			buffer.Reset()
+			logs, err := j.retrieveLogs(ctx, name)
+			if err != nil {
+				return err
+			}
+			logs = logs[skip:]
+			skip += len(logs)
+			if len(logs) == 0 {
+				continue
+			}
+			for _, log := range logs {
+				_, err = fmt.Fprintf(&buffer, "%s [%s:%s] %s\n", time.Unix(log.TS, 0), log.RS, log.Node, log.Msg)
+				if err != nil {
+					return err
+				}
+			}
+			send(&agentpb.JobProgress{
+				JobId:     j.id,
+				Timestamp: ptypes.TimestampNow(),
```

Comment:
why timestamp is not taken from the original message?

Reply:
What means original message?
It's time of when progress was sent. Also I think in MySQL backup logs, we don't have time for each line.

Reply:
which timestamp the end-user will se in result? one when the log record was produced or one that we set it (in this line) ?

Reply:
End-user actual timestamp it was produced. (here in mongo pbm)

Reply:
Right now for nothing, I guess. Maybe @artemgavrilov could clarify?

Reply:
This is a message timestamp, all messages have it. It's not used anywhere right now, but it may be helpful for debugging.